src/include/Context.h

```diff
@@ -58,6 +59,7 @@ class Context {
   virtual void finish(int r) = 0;
 
  public:
+  static const int NO_PGLOCK = 256;
```

Comment:
I just skipped PG lock when ShardedOpWQ holds PG lock.
So, I think the above approach is safe.

If needed, i separate the patch into two